runmariadb.sh

```diff
@@ -47,7 +47,10 @@ EOSQL
 
     # mysqld_safe closes stdout/stderr if no bash options are set ($- == '')
     # turn on tracing to prevent this
-    exec bash -x /usr/bin/mysqld_safe --init-file /tmp/configure-mysql.sql
+    exec bash -x /usr/bin/mysqld_safe --init-file /tmp/configure-mysql.sql &
```

Comment:
This will prevent stdout from receiving the logs, as far as I understand.

Reply:
Even though it's backgrounded it should still keep the same stdout/stdin/stderr. We can take a look at the CI logs once it finishes.

Reply:
Looks like it has logs

```
+ logging=init
+ parse_arguments --init-file /tmp/configure-mysql.sql
++ echo --init-file
+ val=--init-file
+ append_arg_to_args --init-file
++ shell_quote_string --init-file
++ echo --init-file
+ args=' --init-file'
+ args=' --init-file /tmp/configure-mysql.sql'
+ args=' --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log'
+ args=' --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log --pid-file=/run/mariadb/mariadb.pid'
+ args=' --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log --pid-file=/run/mariadb/mariadb.pid --socket=/var/lib/mysql/mysql.sock'
+ cmd='nohup /usr/libexec/mysqld  --basedir=/usr --datadir=/var/lib/mysql --plugin-dir=/usr/lib64/mariadb/plugin --user=mysql  --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log --pid-file=/run/mariadb/mariadb.pid --socket=/var/lib/mysql/mysql.sock'
+ cmd='nohup /usr/libexec/mysqld  --basedir=/usr --datadir=/var/lib/mysql --plugin-dir=/usr/lib64/mariadb/plugin --user=mysql  --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log --pid-file=/run/mariadb/mariadb.pid --socket=/var/lib/mysql/mysql.sock < /dev/null'
+ eval_log_error 'nohup /usr/libexec/mysqld  --basedir=/usr --datadir=/var/lib/mysql --plugin-dir=/usr/lib64/mariadb/plugin --user=mysql  --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log --pid-file=/run/mariadb/mariadb.pid --socket=/var/lib/mysql/mysql.sock < /dev/null'
+ local 'cmd=nohup /usr/libexec/mysqld  --basedir=/usr --datadir=/var/lib/mysql --plugin-dir=/usr/lib64/mariadb/plugin --user=mysql  --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log --pid-file=/run/mariadb/mariadb.pid --socket=/var/lib/mysql/mysql.sock < /dev/null'
+ eval 'nohup /usr/libexec/mysqld  --basedir=/usr --datadir=/var/lib/mysql --plugin-dir=/usr/lib64/mariadb/plugin --user=mysql  --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log --pid-file=/run/mariadb/mariadb.pid --socket=/var/lib/mysql/mysql.sock < /dev/null'
++ nohup /usr/libexec/mysqld --basedir=/usr --datadir=/var/lib/mysql --plugin-dir=/usr/lib64/mariadb/plugin --user=mysql --init-file /tmp/configure-mysql.sql --log-error=/var/log/mariadb/mariadb.log --pid-file=/run/mariadb/mariadb.pid --socket=/var/lib/mysql/mysql.sock
2020-11-19 14:05:11 0 [Note] InnoDB: Completed initialization of buffer pool

```

Reply:
But we also won't be able to detect if mysql has failed, are we okay with that?

Reply:
Hm, good point - if mysqld fails to ever start, the container won't fail. We'll just sit in the ncat below forever. Maybe that should be refactored to include a timeout and proper exit on failure?

Reply:
I wonder if it would work if we changed it to use stdin instead of a temp file? "--init-file -"